Model/Api/UpdateCart.php

```diff
@@ -203,6 +203,28 @@ public function execute($cart, $add_items = null, $remove_items = null, $discoun
                         // Already sent a response with error, so just return.
                         return false;
                     }
+                    // Adding the same data into immutable quote
+                    // in order to avoid missing addons in parent quote
+                    $addToImmutableQuote = true;
+                    $immutableQuoteItems = $immutableQuote->getItems();
+                    if ($immutableQuoteItems) {
+                        foreach ($immutableQuoteItems as $item) {
+                            if ($item->getSku() == $product->getSku()) {
+                                $addToImmutableQuote = false;
```

Comment:
Do you have any specific case in mind where we need this check or you added it just in case?
Is it possible that via add to cart we change quantity i.e. from 1 to 2, not add new item to cart?

Reply:
Let's remove this check then